src/codecs/pnm/decoder.rs

```diff
@@ -589,11 +633,10 @@ impl<R: Read> PnmDecoder<R> {
     fn read_samples<S: Sample>(&mut self, components: u32, buf: &mut [u8]) -> ImageResult<()> {
         match self.subtype().sample_encoding() {
             SampleEncoding::Binary => {
-                let width = self.header.width();
+                let width = TryInto::<usize>::try_into(self.header.width()).unwrap();
                 let height = self.header.height();
-                let bytecount = S::bytelen(width, height, components)?;
+                let bytecount = S::bytelen(width.try_into().unwrap(), height, components)?;
```

Comment:
Are these not undoing each other? In that case I would delay the conversion to `usize` and reassign it later.